src/hammer-vlsi/synthesis/dc/__init__.py

```diff
@@ -66,6 +66,21 @@ def disable_congestion_map(self) -> None:
         f.write(output)
         f.close()
 
+    def generate_tcl_clock_constraints(self) -> str:
+        """Generate TCL fragments for top module clock constraints."""
+        # TODO(edwardw): move to SynopsysTool
+        output = [] # type: List[str]
+
+        clocks = self.get_clock_ports()
+        for clock in clocks:
+            # TODO: are units in Synopsys always in ns?
+            output.append("create_clock {0} -name {0} -period {1}".format(clock.name, clock.period.value_in_units("ns")))
```

Comment:
Doesn't this need a `[get_ports {0}] ` or something?

Reply:
Apparently not, the first {0} captures the port name...